lib/utilrb/pkgconfig.rb

```diff
@@ -207,16 +216,33 @@ def self.parse_dependencies(string)
                     end
                 end
             end
+            
+            if memo.empty?
+                depth = 1
+            else
+                depth = memo.values.max + 1
+            end
+
+            packages.map do |dep|
+                dep = dep.strip
+                if !memo.key?(dep)
+                    memo[dep] = depth
+                end
+            end
 
             result = packages.map do |dep|
                 dep = dep.strip
+                if memo[dep] < depth
```

Comment:
Why this test ? It looks weird ...

Reply:
> Do you think we should provide the full dependencies for required packages also ?

Yes. I definitely expect each and every object in the chain to be well-formed.

As we discussed, dropping the depth thing to simply use the `memo` as a cache (use an object in `memo` and resolve it otherwise) would actually be a simpler implementation, *and* would provide "proper" PkgCOnfig objects in the complete dependency chain